apps/judicial-system/backend/src/app/modules/case/interceptors/caseFile.interceptor.ts

```diff
@@ -0,0 +1,48 @@
+import { Observable } from 'rxjs'
+import { map } from 'rxjs/operators'
+
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+} from '@nestjs/common'
+
+import {
+  CaseAppealState,
+  CaseFileCategory,
+  isPrisonStaffUser,
+  isPrisonSystemUser,
+  User,
+} from '@island.is/judicial-system/types'
+
+import { Case } from '../models/case.model'
+
+@Injectable()
+export class CaseFileInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<Case> {
+    const request = context.switchToHttp().getRequest()
+    const user: User = request.user
+
+    return next.handle().pipe(
+      map((data: Case) => {
+        if (
+          isPrisonStaffUser(user) ||
+          data.appealState !== CaseAppealState.COMPLETED
```

Comment:
This is too agressive as it will remove all files from defenders in indictment cases.

Reply:
Ah, good catch! I added a 

```
if (isDefenceUser(user)) {
    return data
}
```

i.e. nothing changes for defence users.